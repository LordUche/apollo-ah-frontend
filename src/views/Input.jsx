import React from 'react';
import propTypes from 'prop-types';

export default function Input(props) {
  const {
    name, display, onChange, type
  } = props;
  return (
    <div className="form-group mb-4">
      <label htmlFor={name} className="font-weight-bold">{display}</label>
      <input type={type || 'text'} className="form-control" id={name} name={name} onChange={onChange} />
    </div>
  );
}
Input.propTypes = {
  name: propTypes.string.isRequired,
  display: propTypes.string.isRequired,
  onChange: propTypes.func,
  type: propTypes.string,
};
