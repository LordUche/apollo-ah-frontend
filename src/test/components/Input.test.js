import React from 'react';
import { shallow } from 'enzyme';
import Input from '../../views/Input';

describe('<Input  Test Suite>', () => {
  describe('<Input>', () => {
    function onChange() {
      return true;
    }
    it('It should render Input field well', () => {
      const wrapper = shallow(<Input name="email" onChange={onChange} display="Email" type="text" />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
