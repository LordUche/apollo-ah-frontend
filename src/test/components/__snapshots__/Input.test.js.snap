// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input  Test Suite> <Input> It should render Input field well 1`] = `
<div
  className="form-group mb-4"
>
  <label
    className="font-weight-bold"
    htmlFor="email"
  >
    Email
  </label>
  <input
    className="form-control"
    id="email"
    name="email"
    onChange={[Function]}
    type="text"
  />
</div>
`;
